applications/RANSApplication/custom_elements/data_containers/k_epsilon/epsilon_element_data.cpp

```diff
@@ -79,13 +80,19 @@ void EpsilonElementData<TDim>::CalculateGaussPointData(
 
     using namespace RansCalculationUtilities;
 
+    auto& cl_parameters = this->GetConstitutiveLawParameters();
+    cl_parameters.SetShapeFunctionsValues(rShapeFunctions);
+
+    this->GetConstitutiveLaw().CalculateValue(cl_parameters, EFFECTIVE_VISCOSITY, mKinematicViscosity);
```

Comment:
I don't really like getting the effective viscosity like this (see JC's comment in the base `fluid_constitutive_law.h` definition). I'm facing a similar situation for the new weakly compressible element. Having said so, I think we should discuss making the `GetEffectiveViscosity` method `public` in the `fluid_dynamics_constitutive_law.h` class.

(not for this PR OFC).

Reply:
If we make `GetEffectiveViscosity` a public method, then we have to use down casting of the pointer to access that ryt?

Reply:
That's what I'm afraid of... So far we can continue exploiting the `Calculate` but it's a really weird solution...